src/main/java/org/truffleruby/core/array/BasicArrayMirror.java

```diff
@@ -15,9 +15,8 @@
 
     @Override
     public ArrayMirror extractRange(int start, int end) {
-        ArrayMirror newMirror = newMirror(end - start);
-        copyTo(newMirror, start, 0, end - start);
-        return newMirror;
+        DelegatedArrayStorage newStorage = new DelegatedArrayStorage(getArray(), start, end - start);
+        return new DelegatedArrayMirror(newStorage, this);
```

Comment:
As we discussed, I think extractRange() on basic strategies should still make copies to be safe. And makeStorageShared() should be used explicitly first if we want to share the storage.